### PR TITLE
Fallback `_embedding_bag_backward` and force `sparse=false`.

### PR DESCRIPTION
**Pull request by ysiraichi**
Fix: #6008 

This PR, in addition to https://github.com/pytorch/pytorch/pull/129691, add support for embedding bag calls when its `sparse` parameter is true.

**Problem:** when `sparse=true`, `_embedding_bag_backward` called `at::_sparse_coo_tensor_unsafe_symint` function which returns a sparse tensor. Since PyTorch/XLA does not support sparse tensors, this resulted in a dispatch error (see [the original issue](https://github.com/pytorch/xla/issues/6008)).

**Solution:** although, ideally we should support sparse tensors, in the short-term we decided (in an offline discussion) to fallback to the dense backwards function.

cc @miladm @JackCaoG

### CODE REVIEW

File: torch_xla/csrc/aten_xla_type.cpp [hunk 1]
```diff
@@ -1489,6 +1489,24 @@ XLANativeFunctions::_embedding_bag_forward_only(
...
+  TORCH_LAZY_FN_COUNTER_TIMED_TRACING("xla::");
+  TORCH_WARN(
+      "XLA does not support EmbeddingBag sparse backward function. "
+      "Falling back to the dense function.");
+  return at::native::call_fallback_fn<&xla_cpu_fallback,
+                                      ATEN_OP(_embedding_bag_backward)>::
+      call(grad, indices_, offsets_, offset2bag, bag_size_, max_indices_,
+           num_weights, scale_grad_by_freq, mode, /*sparse=*/false,
+           per_sample_weights_opt, padding_idx);
+}
```
Comment (JackCaoG) [inline torch_xla/csrc/aten_xla_type.cpp:1508]: can we add a test to `test/test_operations.py`, run the embedding bag fwd and bwd and make sure it doesn't crash? this way we can prevent it from regressing.

File: torch_xla/csrc/aten_xla_type.cpp [hunk 2]
```diff
@@ -1489,6 +1489,24 @@ XLANativeFunctions::_embedding_bag_forward_only(
...
 
+at::Tensor XLANativeFunctions::_embedding_bag_backward(
+    const at::Tensor& grad, const at::Tensor& indices_,
+    const at::Tensor& offsets_, const at::Tensor& offset2bag,
+    const at::Tensor& bag_size_, const at::Tensor& max_indices_,
+    int64_t num_weights, bool scale_grad_by_freq, int64_t mode, bool sparse,
+    const std::optional<at::Tensor>& per_sample_weights_opt,
+    int64_t padding_idx) {
+  TORCH_LAZY_FN_COUNTER_TIMED_TRACING("xla::");
+  TORCH_WARN(
```
Comment (bhavya01) [inline torch_xla/csrc/aten_xla_type.cpp:1500]: Should warn about this only if sparse is `true`?

**Review by bhavya01 — APPROVED** [review]

File: torch_xla/csrc/aten_xla_type.cpp (hunk 2 above)
Comment (ysiraichi) [reply torch_xla/csrc/aten_xla_type.cpp:1500]: Ah. You are right!

File: test/test_operations.py [hunk 3]
```diff
@@ -2198,6 +2198,82 @@ def foo(x, is_xla=False):
...
+      fresh = tensor
+      # Maybe move to the specified device.
+      if device is not None:
+        fresh = fresh.to(device)
+      # Clone if not cloned already by the previous device move.
+      if fresh.data_ptr() == tensor.data_ptr():
+        fresh = tensor.clone()
+      # Make this tensor a leaf tensor by detaching and reseting its
+      # requires_grad property.
+      fresh = fresh.detach()
```
Comment (vanbasten23) [inline test/test_operations.py:2229]: Very cool. I wonder why we need to make the tensor a leaf tensor. What would happen if we don't do it?

**Review by vanbasten23 — APPROVED** [review]

File: test/test_operations.py (hunk 3 above)
Comment (ysiraichi) [reply test/test_operations.py:2229]: It's so we can grab its `tensor.grad` and compare.

**Pull request merged**